pkg/innerring/internal/metachain/contracts/meta_test.go

```diff
@@ -0,0 +1,262 @@
+			res := metaI.GetTxExecResult(t, hash)
+			require.Len(t, res.Events, 1)
+			require.Equal(t, "ObjectPut", res.Events[0].Name)
+			notificationArgs := res.Events[0].Item.Value().([]stackitem.Item)
+			require.Len(t, notificationArgs, 3)
+			require.Equal(t, cID[:], notificationArgs[0].Value().([]byte))
+			require.Equal(t, oID, notificationArgs[1].Value().([]byte))
+
+			metaValuesExp := meta.Value().([]stackitem.MapElement)
+			metaValuesGot := notificationArgs[2].Value().([]stackitem.MapElement)
+			require.Equal(t, metaValuesExp, metaValuesGot)
+		})
+
+		t.Run("missing required values", func(t *testing.T) {
+			testFunc := func(key string) {
+				meta := testMeta(cID[:], oID[:])
+				meta.Drop(meta.Index(stackitem.Make(key)))
+				raw, err := stackitem.Serialize(meta)
+				require.NoError(t, err)
+				metaI.InvokeFail(t, fmt.Sprintf("'%s' not found", key), "submitObjectPut", raw, sigs)
+			}
+
+			testFunc("oid")
+			testFunc("size")
+			testFunc("validUntil")
+			testFunc("network")
+		})
+
+		t.Run("incorrect values", func(t *testing.T) {
+			testFunc := func(key string, newVal any) {
+				meta := testMeta(cID[:], oID[:])
+				meta.Add(stackitem.Make(key), stackitem.Make(newVal))
+				raw, err := stackitem.Serialize(meta)
+				require.NoError(t, err)
+				metaI.InvokeFail(t, "incorrect", "submitObjectPut", raw, sigs)
+			}
+
+			testFunc("oid", []byte{1})
+			testFunc("validUntil", 1) // tested chain will have some blocks for sure
+			testFunc("network", netmode.UnitTestNet+1)
+			testFunc("type", math.MaxInt64)
+			testFunc("firstPart", []byte{1})
+			testFunc("previousPart", []byte{1})
+			testFunc("deleted", []any{[]byte{1}})
+			testFunc("locked", []any{[]byte{1}})
+		})
+	})
+}
+
+func testMeta(cid, oid []byte) *stackitem.Map {
+	deleted := oidtest.ID()
+	locked := oidtest.ID()
+
+	return stackitem.NewMapWithValue(
+		[]stackitem.MapElement{
+			{Key: stackitem.Make("network"), Value: stackitem.Make(netmode.UnitTestNet)},
+			{Key: stackitem.Make("cid"), Value: stackitem.Make(cid)},
+			{Key: stackitem.Make("oid"), Value: stackitem.Make(oid)},
+			{Key: stackitem.Make("firstPart"), Value: stackitem.Make(oid)},
+			{Key: stackitem.Make("size"), Value: stackitem.Make(123)},
+			{Key: stackitem.Make("deleted"), Value: stackitem.Make([]any{deleted[:]})},
+			{Key: stackitem.Make("locked"), Value: stackitem.Make([]any{locked[:]})},
+			{Key: stackitem.Make("validUntil"), Value: stackitem.Make(math.MaxInt)},
+		})
+}
+
+func newCustomNatives(cfg neogoconfig.ProtocolConfiguration) []interop.Contract {
+	var (
+		defaultContracts = native.NewDefaultContracts(cfg)
+		newContracts     = make([]interop.Contract, 0)
+
+		neoContract native.INEO
+	)
+	for _, contract := range defaultContracts {
+		switch contract.(type) {
+		case *native.NEO:
+			neoContract = contract.(native.INEO)
+			newContracts = append(newContracts, neoContract)
+		case *native.Management, *native.Ledger, *native.GAS, *native.Policy, *native.Designate, *native.Notary:
```

Comment:
We're supposed to have a custom (and simplified) Neo/Gas implementations. Also, do you need Notary at all for this chain?

Reply:
I suppose we need a separate issue to simplify the set of native contracts for meta chain.